src/execution_plan/ops/op_update.c

```diff
@@ -149,25 +161,64 @@ static Record _handoff(OpUpdate *op) {
 	return NULL;
 }
 
-OpBase *NewUpdateOp(const ExecutionPlan *plan, EntityUpdateEvalCtx *update_exps) {
+static void _groupUpdateExps(OpUpdate *op, EntityUpdateEvalCtx *update_ctxs) {
+	// Sort update contexts by unique Record ID.
+#define islt(a,b) (a->record_idx < b->record_idx)
+
+	uint n = array_len(update_ctxs);
+	QSORT(EntityUpdateEvalCtx, update_ctxs, n, islt);
+
+	/* Create an EntityUpdateCtx for each alias that will be modified by this op.
+	 * Group all relevant EvalCtx structs on each new context. */
+	op->update_ctxs = array_new(EntityUpdateCtx, 1);
+
+	EntityUpdateEvalCtx *prev = NULL;
+	EntityUpdateCtx *entity_ctx = NULL;
+
+	for(uint i = 0; i < n; i++) {
+		EntityUpdateEvalCtx *current = &update_ctxs[i];
+		if(!prev || current->record_idx != prev->record_idx) {
+			// Encountered a diffrent entity being updated; create a new context.
+			EntityUpdateCtx new_ctx = { .alias = current->alias,
+										.record_idx = current->record_idx,
+										.updates = array_new(PendingUpdateCtx, 1),
+										.exps = array_new(EntityUpdateEvalCtx, 1),
+									  };
```

Comment:
On GitHub, the lines offset is not the same. If in the code it looks good, let it be

Reply:
GitHub prints tabs as 8 spaces rather than 4 for some reason, but this looks normal in an editor!